main.m

```diff
@@ -373,7 +373,13 @@ void printGenericPassword(NSDictionary *passwordItem)
 	printToStdOut(@"Synchronizable: %@\n", [passwordItem objectForKey:(id)kSecAttrSynchronizable]); 
 	printToStdOut(@"Generic Field: %@\n", [[passwordItem objectForKey:(id)kSecAttrGeneric] description]); 
 	NSData* passwordData = [passwordItem objectForKey:(id)kSecValueData];
-	printToStdOut(@"Keychain Data: %@\n\n", [[NSString alloc] initWithData:passwordData encoding:NSUTF8StringEncoding]);
+	NSString *keychainData = [[NSString alloc] initWithData:passwordData encoding:NSUTF8StringEncoding];
+	if (keychainData) {
+		printToStdOut(@"Keychain Data: %@\n\n", keychainData);
+	}
+	else {
```

Comment:
Not a huge deal..but is there a cleaner way to do this than just assume a falsey `kechainData` means that it wasn’t UTF-8 data? For example, can we check the encoding and then condition the if/else off of which option we use to print the data?

Reply:
Are there any other error conditions that could cause a piece of data to not be convertible to a string?  I think "contains invalid bytes" is the only one.
The import of `initWithData:encoding:` in Swift is `init?(data: Data, encoding: String.Encoding)`, a failable initializer (which is generally used for things that can only fail in one way in comparison with a throwing initializer, for things that can fail in multiple ways), and the expected way to deal with this in Swift is identical (well Swift has `if let` as a shortcut for the assignment + null check but other than that identical), so I think this *is* the expected way to do this check.